Notation.Plugin.AzureKeyVault/Command/GenerateSignature.cs

```diff
@@ -0,0 +1,59 @@
+using System.Text.Json;
+using Notation.Plugin.AzureKeyVault.Certificate;
+using Notation.Plugin.AzureKeyVault.Client;
+using Notation.Plugin.Protocol;
+
+namespace Notation.Plugin.AzureKeyVault.Command
+{
+    /// <summary>
+    /// Implementation of describe-key command.
+    /// </summary>
+    public class GenerateSignature : IPluginCommand
+    {
+        public async Task<object> RunAsync(string inputJson)
+        {
+            // Parse the input
+            GenerateSignatureRequest? input = JsonSerializer.Deserialize<GenerateSignatureRequest>(inputJson);
+            if (input == null)
+            {
+                throw new ValidationException("Invalid input");
+            }
+
+            var akvClient = new KeyVaultClient(input.KeyId);
+
+            // Extract signature algorithm from the certificate
+            var leafCert = await akvClient.GetCertificate();
+            var keySpec = KeySpecUtils.ExtractKeySpec(leafCert);
+            var signatureAlgorithm = SignatureAlgorithmHelper.FromKeySpec(keySpec);
+
+            // Sign
+            var signature = await akvClient.Sign(signatureAlgorithm, input.Payload);
+
+            // Build the certificate chain
+            List<byte[]> certificateChain = new List<byte[]>();
+            if (input.PluginConfig != null && input.PluginConfig.ContainsKey("ca_certs"))
+            {
+                // Build the entire certificate chain from the certificate 
+                // bundle (including the intermediate and root certificates).
+                var caCertsPath = input.PluginConfig["ca_certs"];
+                certificateChain = CertificateChain.Build(CertificateBundle.Create(caCertsPath), leafCert);
+            }
+            else if (input.PluginConfig != null && input.PluginConfig.ContainsKey("as_secret"))
+            {
+                // Read the entire certificate chain from the Azure Key Vault with GetSecret permission.
+                throw new NotImplementedException("as_secret is not implemented yet");
+            }
+            else
+            {
+                // Self-signed leaf certificate
+                certificateChain.Add(leafCert.RawData);
```

Comment:
IMO, we should validate that the cert is indeed self-signed. If not, we should throw an error.

Reply:
Added the validation. If the this is not a self-signed certificate the error message will return:
> Failed to build the X509 chain up to the root certificate. To resolve this issue, provide the intermediate and root certificates by passing the certificate bundle file's path to the `ca_certs` key in the pluginConfig